tests/tensor/test_math_opt.py

```diff
@@ -4544,3 +4544,53 @@ def logit_fn(x):
     fg = optimize(FunctionGraph([x], [out]))
     assert not list(fg.toposort())
     assert fg.inputs[0] is fg.outputs[0]
+
+
+class TestDistributiveOpts:
+    x_at = vector("x")
+    y_at = vector("y")
+    a_at = matrix("a")
+
+    # x_lv, if any, is the logic variable version of some term x,
+    # while x_at is the Aesara tensor version for the same.
+    @pytest.mark.parametrize(
+        "orig_operation, optimized_operation",
+        [
+            (a_at * x_at + a_at * y_at, a_at * (x_at + y_at)),
+            (x_at * a_at + y_at * x_at, (x_at + y_at) * a_at),
```

Comment:
So I managed to generalize them for `div` and `mul` but it seems that these rewrites aren't taking the `facts()` into account. For instance this particular case passes but the case above it doesn't. Isn't this supposed to be handled by `fact(commutative, at.mul)` in the rewrite.

Reply:
This might have to do with the `eq` goal that's being used at specific steps (even behind the scenes).  There are special `eq_*` goals that do (and don't) take into account the associativity/commutativity (AC) information set by `facts`, and you may need to use those explicitly at certain points.  Just try not to use them when they're not needed; otherwise, the streams resulting from the goals will become very long due to all the permutations induced by AC relations.